apps/notifications/views.py

```
@@ -0,0 +1,24 @@
+
+class NotificationViewSet(viewsets.ReadOnlyModelViewSet):
+    """Provide read access to notifications with actions to manage state."""
+
+    serializer_class = NotificationSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    def get_queryset(self):
+        return Notification.objects.filter(recipient=self.request.user)
+
+    @action(detail=True, methods=['post'])
+    def mark_read(self, request, pk=None):
```

Ruff (0.14.0)

Unused method argument: `request`

(ARG002)